graphql-server/pkg/queries/proposal.go

```diff
@@ -0,0 +1,72 @@
+package queries
+
+import (
+	"context"
+	"strconv"
+
+	"github.com/oursky/likedao/pkg/config"
+	"github.com/oursky/likedao/pkg/models"
+	"github.com/pkg/errors"
+	"github.com/uptrace/bun"
+)
+
+type IProposalQuery interface {
+	QueryProposalConnection(input models.QueryProposalsInput) (*models.ProposalConnection, error)
```

Comment:
best to decouple gql schema from db level utils; how about adding ScopeXXX filters instead of accepting QueryProposalsInput here?

e.g.:
```
q := pkgContext.GetQueriesFromCtx(ctx).Proposal
res, err := q.ScopeStatus(input.Filter.ToStatus()).ScopeSearchTerm(input.SearchTerm).ScopeAddress(input.Address).Paginate(input.after, input.first).QueryProposals()
```

& instead of returning connection directly, how about decouple & introduce a new QueryProposalsResult struct to return array of model type + pagination info?

Reply:
so gql related models are only used at resolver level; & we can reuse db level methods easily in the future

Reply:
actually, how about `QueryPaginatedProposals(input.after, input.first)`? to return paginated result; `QueryProposals` should simply give array of model.Proposals